dm/pkg/utils/db.go

```diff
@@ -632,3 +632,23 @@ func GetTableCreateSQL(ctx context.Context, conn *sql.Conn, tableID string) (sql
 	}
 	return createStr, nil
 }
+
+// GetMaxConnections gets max_connections for sql.DB which is suitable for session variable max_connections.
+func GetMaxConnections(ctx context.Context, db *sql.DB) (int, error) {
+	c, err := db.Conn(ctx)
+	if err != nil {
+		return 0, err
+	}
+	defer c.Close()
+	return GetMaxConnectionsForConn(ctx, c)
+}
+
+// GetMaxConnectionsForConn gets max_connections for sql.Conn which is suitable for session variable max_connections.
+func GetMaxConnectionsForConn(ctx context.Context, conn *sql.Conn) (int, error) {
```

Comment:
seems only `GetMaxConnections` used `GetMaxConnectionsForConn` why split this to two func ?

Reply:
I'm just imitating `GetParserForConn`.